File: LeetCode/Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
       map<char, int> values = {{'I', 1}, {'V', 5},{'X', 10},{'L', 50},
  {'C', 100},{'D', 500},{'M', 1000}};
  
  int total = 0;
  for(int i = 0; i < s.length(); i++){
    
    if(values[s[i]] >= values[s[i+1]]){
      total = total + values[s[i]]; 
    }

    else{
      total = total - values[s[i]];  
    }
  }
  return total;
    }
};

File: Coding Platforms(codechef,codeforces,etc)/LeetCode/roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        n = len(s)
        num = 0
        for i in range(n):
            if i+1<n and roman[s[i]]<roman[s[i+1]]:
                num -= roman[s[i]]
            else:
                num += roman[s[i]]
        return num
